### ISSUE TITLE
$0 is show full path when use as global bin

### ISSUE THREAD

**Issue by atian25** [yargs/yargs#811]
- use [nvs](https://github.com/jasongin/nvs)
- create a simple bin call `single-bin`
  - `usage = 'Usage: $0 --baseDir=<target dir>';`
- `npm link` to install it as global bin for test
- exec `single-bin --help`

```bash
➜  single-bin --help
Usage: /Users/tz/git/nvs/default/bin/single-bin --baseDir=<target dir>
```

`/Users/tz/git/nvs/default/bin/single-bin` is expect to be `single-bin`

**Comment by atian25** [comment]
```js
    self.$0 = process.env._.replace(
      path.dirname(process.execPath) + '/', ''
    )
```
- `process.env._ = /Users/tz/git/nvs/default/bin/my-bin`
- `process.execPath = /Users/tz/git/nvs/node/6.10.0/x64/bin/node`

**Comment by jasongin** [comment]
```
    self.$0 = process.env._.replace(
      path.dirname(process.execPath) + '/', ''
    )
```
This code is relying on the assumption that the executable is under a non-symlinked path, which is not a very safe assumption. You could reproduce this problem without NVS by installing the test project under a path that includes a directory symlink (instead of the npm global bin).

**Comment by rmg** [comment]
The `execPath` is expanded to the `realpath` following the same semantics as node module loading.

One solution could be to ensure that both paths are equally "real" with something like this:
```js
self.$0 = fs.realPathSync(process.env._).replace(
  path.dirname(process.execPath) + '/', ''
)
```

**Comment by atian25** [comment]
@rmg

seems nvs path is special

- `fs.realPathSync(process.env._) === /Users/tz/git/nvs/node/6.10.0/x64/lib/node_modules/my-git/bin/my-git.js`
- `path.dirname(process.execPath) === /Users/tz/git/nvs/node/6.10.0/x64/bin`

**Comment by jasongin** [comment]
The `fs.realPathSync()` call doesn't easily solve the problem because it resolves *two* symbolic links in the path: `default -> node/6.10.0/x64` and `my-git` -> `../lib/node_modules/my-git/bin/my-git.js`. For the logic in this code to work, the filename symlink should not be resolved:
```JavaScript
var realPath = path.join(fs.realPathSync(path.dirname(process.env._)), path.basename(process.env._))
self.$0 = realPath.replace(path.dirname(process.execPath) + '/', '')
```

**Comment by rmg** [comment]
Good catch, @jasongin.

If I were using `yargs` in an app this is the point where I would just set `$0` statically myself in my CLI.

**Comment by laggingreflex** [comment]
Fixed in #1143. You can now set it with `.scriptName()`
